docs-src/tls.md

```diff
@@ -0,0 +1,328 @@
+- **Upstream**: This is the connection between the Gateway and Service. The
+  Service in question can be a core Kubernetes Service or any other network
+  endpoint as well.
+  
+With Service APIs, TLS configuration of downstream and
+upstream connections is managed independently.
+
+Please note that in case of `Passthrough` TLS mode, no TLS settings take
+effect as the TLS session from the client is NOT terminated at the Gateway.
+The rest of the document assumes that TLS is being terminated at the Gateway.
+
+## Downstream TLS
+
+Downstream TLS settings are configured using Listeners at the Gateway level.
+
+### Listener and TLS 
+
+Listeners expose the TLS setting on a per domain or sub-domain basis.
+TLS settings of a listener are applied to all domains that satisfy the
+`hostname.match` criteria.
+
+For example:
+
+```yaml
+listeners:
+- protocol: HTTPS # Other possible value is `TLS`
+  port: 443
+  hostname:
+    match: Any # Other possible values are `Domain` and `Exact`
+  tls:
+    mode: Terminate
+    certificateRef:
+      kind: Secret
+      group: core
+      name: default-cert
+    routeOverride:
+      certificate: Deny
+```
+
+provides a TLS certificate for any request that is served a Gateway.
+If `hostname.match` is set to `Exact`, then the TLS settings apply to only
+the specific hostname that is set in `hostname.name`
+This is the recommended way of configuring TLS settings in Service APIs
+as it does not involve any redirection and should suffice in majority of the
+use-cases.
+
+For more clarity, please take a look at the examples below.
+
+### Route and TLS
+
+If `listener.tls.routeOverride.certificate` is set to `Allow`, TLS certificates
+can be configured at the Route-level.
+This feature is primarily  meant for a cluster with a self-service model where
+Service owners bring their own TLS certificates. This feature also mirrors
+the behavior of TLS as defined in the Ingress v1 resource.
+One should use this feature only when the Gateway Admin wishes to delegate
+TLS to Service or App owners.
+
+When using this feature, please note that TLS certificate to serve is chosen
+before an HTTPRoute is selected. This is because TLS handshake is completed
+before an HTTP request is sent from the client.
+
+[TLS Certificate in Route](#tls-certificate-in-route) provides an example
+on how this feature can be used.
+
+### Examples
+
+#### TLS in listener
+
+In this example, Gateway is configured to serve `foo.example.com` and
+`bar.example.com` domains. The certificate for these domains are specified
+in Gateway.
+
+```yaml
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - protocol: HTTPS
+    port: 443
+    hostname:
+      name: foo.example.com
+      match: Exact
+    tls:
+      certificateRef:
+        kind: Secret
+        group: core
+        name: foo-example-com-cert
+      routeOverride:
+        certificate: Deny
```

Comment:
I considered this but decided against it because of two reasons:
- makes it clear that TLS config is ONLY in Listener
- for anybody trying to learn the TLS section of this API, it helps with keeping all bits in front. I've already remoed route selection section. 

That being said, I can reason either ways and don't feel strongly about it.